File: simplesplit-backend/src/main/scala/com/github/ai/split/entity/api/request/PostGroupRequest.scala
package com.github.ai.split.entity.api.request

import zio.json.{DeriveJsonDecoder, JsonDecoder}

case class PostGroupRequest(
  password: String,
  title: String,
  description: Option[String]
)

object PostGroupRequest {
  implicit val decoder: JsonDecoder[PostGroupRequest] = DeriveJsonDecoder.gen[PostGroupRequest]
}

File: android/backend-api/src/main/kotlin/com/github/ai/split/api/request/PostGroupRequest.kt
package com.github.ai.split.api.request

import kotlinx.serialization.Serializable
import com.github.ai.split.api.NewExpenseDto
import com.github.ai.split.api.UserNameDto

@Serializable
data class PostGroupRequest(
    val password: String,
    val title: String,
    val description: String?,
    val currencyIsoCode: String,
    val members: List<UserNameDto>?,
    val expenses: List<NewExpenseDto>?
)